worlds/oot_soh/location_access/dungeons/jabujabus_belly.py

```diff
@@ -4,6 +4,233 @@
+    ])
+
+    ## Jabu Jabu's Belly Bigocto Room
+    # Locations
+    add_locations(Regions.JABU_JABUS_BELLY_BIGOCTO_ROOM, world, [
+        (Locations.JABU_JABUS_BELLY_GS_LOBBY_BASEMENT_UPPER, lambda bundle: is_adult(bundle) and can_get_enemy_drop(bundle, Enemies.GOLD_SKULLTULA, EnemyDistance.SHORT_JUMPSLASH))
+    ])
+    # Connections
+    connect_regions(Regions.JABU_JABUS_BELLY_BIGOCTO_ROOM, world, [
+        (Regions.JABU_JABUS_BELLY_B1_NORTH, lambda bundle: True),
+        (Regions.JABU_JABUS_BELLY_ABOVE_BIGOCTO, lambda bundle: has_item(LocalEvents.JABU_JABUS_BELLY_RUTO_IN_1F_RESCUED, bundle, world) and can_kill_enemy(bundle, Enemies.BIG_OCTO))
+    ])
+
+    ## Jabu Jabu's Belly Above Bigocto
+    # Events (TODO: Add FairyPot and NutPot events)
+    add_events(Regions.JABU_JABUS_BELLY_ABOVE_BIGOCTO, world, [
+        (EventLocations.JABU_JABUS_BELLY_ABOVE_BIGOCTO_FAIRY_POT, Events.CAN_ACCESS_FAIRIES, lambda bundle: True),
+        (EventLocations.JABU_JABUS_BELLY_ABOVE_BIGOCTO_NUT_POT, Events.CAN_FARM_NUTS, lambda bundle: True)
+    ])
+    # Locations
+    add_locations(Regions.JABU_JABUS_BELLY_ABOVE_BIGOCTO, world, [
+        (Locations.JABU_JABUS_BELLY_ABOVE_BIG_OCTO_POT_1, lambda bundle: can_break_pots(bundle)),
+        (Locations.JABU_JABUS_BELLY_ABOVE_BIG_OCTO_POT_2, lambda bundle: can_break_pots(bundle)),
+        (Locations.JABU_JABUS_BELLY_ABOVE_BIG_OCTO_POT_3, lambda bundle: can_break_pots(bundle))
+    ])
+    # Connections
+    connect_regions(Regions.JABU_JABUS_BELLY_ABOVE_BIGOCTO, world, [
+        (Regions.JABU_JABUS_BELLY_LIFT_UPPER, lambda bundle: can_use(Items.BOOMERANG, bundle))
+    ])
+
+    ## Jabu Jabu's Belly Lift Upper
+    # Events
+    add_events(Regions.JABU_JABUS_BELLY_LIFT_UPPER, world, [
+        (EventLocations.JABU_JABUS_BELLY_LOWERED_PATH, LocalEvents.JABU_JABUS_BELLY_LOWERED_PATH_ACTIVATED, lambda bundle: True)
+    ])
+    # Connections
+    connect_regions(Regions.JABU_JABUS_BELLY_LIFT_UPPER, world, [
+        (Regions.JABU_JABUS_BELLY_MAIN, lambda bundle: True)
+    ])
+
+    ## Jabu Jabu's Belly Near Boss Room
+    # Locations
+    add_locations(Regions.JABU_JABUS_BELLY_NEAR_BOSS_ROOM, world, [
+        (Locations.JABU_JABUS_BELLY_GS_NEAR_BOSS, lambda bundle: can_kill_enemy(bundle, Enemies.GOLD_SKULLTULA, EnemyDistance.BOMB_THROW))
+    ])
+    # Connections
+    connect_regions(Regions.JABU_JABUS_BELLY_NEAR_BOSS_ROOM, world, [
+        (Regions.JABU_JABUS_BELLY_MAIN, lambda bundle: True),
+        (Regions.JABU_JABUS_BELLY_BOSS_ENTRYWAY, lambda bundle: can_use(Items.BOOMERANG, bundle) or (can_do_trick(Tricks.JABU_NEAR_BOSS_RANGED, bundle) and (can_use(Items.HOOKSHOT, bundle) or can_use(Items.FAIRY_BOW, bundle) or can_use(Items.FAIRY_SLINGSHOT, bundle))) or (can_do_trick(Tricks.JABU_NEAR_BOSS_EXPLOSIVES, bundle) and (can_use(Items.BOMBCHU_5, bundle) or (can_use(Items.HOVER_BOOTS, bundle) and can_use(Items.BOMB_BAG, bundle)))))
+    ])
```

Comment:
could be simplified with can_use_any
```suggestion
        (Regions.JABU_JABUS_BELLY_BOSS_ENTRYWAY, lambda bundle: can_use(Items.BOOMERANG, bundle) or (can_do_trick(Tricks.JABU_NEAR_BOSS_RANGED, bundle) and (can_use_any([Items.HOOKSHOT, Items.FAIRY_BOW, Items.FAIRY_SLINGSHOT], bundle))) or (can_do_trick(Tricks.JABU_NEAR_BOSS_EXPLOSIVES, bundle) and (can_use(Items.BOMBCHU_5, bundle) or (can_use(Items.HOVER_BOOTS, bundle) and can_use(Items.BOMB_BAG, bundle)))))
    ])
```